Enable shortcuts to package names

zivkaziv opened the pull request:
In case of package name longer than 2 chars, enable shortcuts to package names

giltayar commented:
Good stuff as a start! Shouldn't be long before this is complete. Could you please also add to the documentation in the `.md` files?

giltayar commented:
@zivkaziv, please merge main branch because I fixed a test in `npm-teskit` which fails the CI.

zivkaziv commented:
Sorry @giltayar that it took me forever to continue my PR.
I've changed the `.find` to `.filter` so in case we have more than 1 matching package we will throw `Error`
Regarding the unit test, I didn't really understand what I did... I fixed it:) 
I wanted to add two more tests for the following cases:
- More than one package found
- No package found

Something like
```
    await expect(runBuild(cwd, `more then one package - don't build anything`, ['package'], undefined)).to.throw(/there are 3 packages with the name/);
    await expect(runBuild(cwd, `no package match the name - don't build anything`, ['bilt'], undefined)).to.throw(/cannot find a package with the name/);
```
But it didn't work... I'll be happy for some help with those tests:)

Regarding the `.md` file, I'm not sure as well it covers this feature...
Let me know what do you think

Thanks
Ziv

giltayar commented:
@zivkaziv regarding the tests:

> `await expect(runBuild(cwd, `more then one package - don't build anything`, ['package'], undefined)).to.throw(/there are 3 packages with the name/);`

It should be `await expect(async () => runBuild(...)).to.thow(...)`, but unfortunately, Chai doesn't support promises out of the box. You have two options:

1. Install and use the `chai-as-promised` extension
2. My personal favorite: do a catch yourself, return it, and check it:

```js
expect(await runBuild(...).catch(err => err.message)).to.include('there are 3 packages')
```

giltayar commented on docs/reference.md:
```diff
@@ -106,7 +106,7 @@ are relevant for _all_ build job configurations.
 * `--config`: the config file to read. Optional, and if does not exist, will search
   for it, as described [above](#configuration-file).
-* `packagesToBuild`: a set of package directories or package names. If a directory, it MUST
+* `packagesToBuild`: a set of package directories or shortcuts or the entire package names. If a directory, it MUST
```
A bit laconic as information. More information would be appreciated. It doesn't have to be in this paragraph. It could be in a separate paragraph underneath it).

zivkaziv replied:
Updated:)

giltayar reviewed (changes requested):
Finish the test, and add more information to the documentation, and we're good to go!

zivkaziv commented:
Thank you:) Your comments really helped...

giltayar commented:
@zivkaziv can you resolve the conflicts by rebasing on the main branch? Note that conflicts in `package-lock.json` can remain (i.e. you can resolve the file with the conflict markers), because `npm install` knows how to clean them.

Once you've done that, we can merge (assuming all tests pass!). 💪

zivkaziv commented:
Tests are passed and conflicts  resolved...
```
**** [a] build package succeeded
    ✓ should disable all git stuff when --no-git, and test all build options (27351ms)


  20 passing (3m)
```
It's always exciting to work on open source... Thank you

giltayar commented:
🎉🎉🎉
